kasa/interfaces/lightpreset.py

```diff
@@ -0,0 +1,89 @@
+"""Module for LightPreset base class."""
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Sequence
+
+from ..feature import Feature
+from ..module import Module
+from .light import LightState
+
+
+class LightPresetInfo(ABC):
+    """Light configuration preset."""
+
+    brightness: int
+
+    hue: int | None
+    saturation: int | None
+    color_temp: int | None
+
+
+class LightPreset(Module):
+    """Base interface for light preset module."""
+
+    PRESET_NOT_SET = "Not set"
+
+    def _initialize_features(self):
+        """Initialize features."""
+        device = self._device
+        self._add_feature(
+            Feature(
+                device,
+                id="light_preset",
+                name="Light preset",
+                container=self,
+                attribute_getter="preset",
+                attribute_setter="set_preset",
+                category=Feature.Category.Config,
+                type=Feature.Type.Choice,
+                choices_getter="preset_list",
+            )
+        )
+
+    @property
+    @abstractmethod
+    def preset_list(self) -> list[str]:
+        """Return built-in effects list.
+
+        Example:
+            ['Off', 'Preset 1', 'Preset 2', ...]
+        """
+
+    @property
+    @abstractmethod
+    def preset_states_list(self) -> Sequence[LightState]:
+        """Return built-in effects list.
+
+        Example:
+            ['Off', 'Preset 1', 'Preset 2', ...]
+        """
```

Comment:
Will fix the docstring.  The single `preset_list(self) -> dict` approach does not play nice with the `Feature.Type.Choice` approach of having a list of names that can be chosen.  This is also consistent with the `LightEffect` module.

Reply:
Hmm, could we extend the way we handle choices instead? Just an idea, I'm fine with this approach, too.

Reply:
It might be an idea to make `choices` a property calls the getter each time as I think at the moment it might not update after the values changes.  Will look at that in a subsequent PR.